src/Version/Service/VersionBinaryService.php

```diff
@@ -127,4 +121,18 @@ public function streamPdfPreview(
 
         return $this->documentService->getPreviewStream($document);
     }
+
+    private function getImageThumbnail(Image $image): ThumbnailInterface
+    {
+        $image->setFilename(uniqid('', true));
```

Comment:
should the filename maybe the the original filename + unique?

Reply:
And maybe we should reset the filename to the original one afterwards so that calling this service does not change the state of the asset instance?

Reply:
I could not reproduce the issue ...but is it caused because of the response headers or because of the asset instance filename?

Reply:
Because of the asset instance filename - also with disabled browser cache the same happens. I agree that it's not easy to reproduce without frontend.